Mage/src/main/java/mage/abilities/effects/common/ExileThenReturnTargetEffect.java

```diff
@@ -50,15 +49,15 @@ public ExileThenReturnTargetEffect copy() {
     @Override
     public boolean apply(Game game, Ability source) {
         Player controller = game.getPlayer(source.getControllerId());
-        Set<Card> toFlicker = new LinkedHashSet<>();
```

Comment:
Correct replacement of LinkedHashSet by streams (don't use toSet, cause it can random cards order):
https://stackoverflow.com/a/51945053

Reply:
Why is the order necessary?

Reply:
Its related to UX. Example: users must see same logs and same GUI requests for same actions. LinkedHashSet helps to save that order.

Reply:
Hmm, well, this is probably a much broader issue in the codebase then (indiscriminate use of `HashSet<>` rather than `LinkedHashSet<>` in addition to the `Collectors.toSet()`). I'll fix it here at least.